ml-algorithms/src/main/java/org/opensearch/ml/engine/tools/IndexMappingTool.java

```diff
@@ -86,75 +88,80 @@ public Object parse(Object o) {
-                        sb.append("index: ").append(index).append("\n\n");
-
-                        MappingMetadata mapping = getIndexResponse.mappings().get(index);
-                        if (mapping != null) {
-                            sb.append("mappings:\n");
-                            for (Entry<String, Object> entry : mapping.sourceAsMap().entrySet()) {
-                                sb.append(entry.getKey()).append("=").append(entry.getValue()).append('\n');
-                            }
-                            sb.append("\n\n");
+            ActionListener<GetIndexResponse> internalListener = new ActionListener<GetIndexResponse>() {
+
+                @Override
+                public void onResponse(GetIndexResponse getIndexResponse) {
+                    try {
+                        // Handle empty response
+                        if (getIndexResponse.indices().length == 0) {
+                            @SuppressWarnings("unchecked")
+                            T empty = (T) ("There were no results searching the index parameter [" + parameters.get("index") + "].");
+                            listener.onResponse(empty);
+                            return;
                         }
+                        StringBuilder sb = new StringBuilder();
+                        for (String index : getIndexResponse.indices()) {
+                            sb.append("index: ").append(index).append("\n\n");
+
+                            MappingMetadata mapping = getIndexResponse.mappings().get(index);
+                            if (mapping != null) {
+                                sb.append("mappings:\n");
+                                for (Entry<String, Object> entry : mapping.sourceAsMap().entrySet()) {
+                                    sb.append(entry.getKey()).append("=").append(entry.getValue()).append('\n');
+                                }
+                                sb.append("\n\n");
+                            }
 
-                        Settings settings = getIndexResponse.settings().get(index);
-                        if (settings != null) {
-                            sb.append("settings:\n").append(settings.toDelimitedString('\n')).append("\n\n");
+                            Settings settings = getIndexResponse.settings().get(index);
+                            if (settings != null) {
+                                sb.append("settings:\n").append(settings.toDelimitedString('\n')).append("\n\n");
+                            }
                         }
-                    }
 
-                    @SuppressWarnings("unchecked")
-                    T response = (T) sb.toString();
-                    listener.onResponse(response);
-                } catch (Exception e) {
-                    onFailure(e);
+                        @SuppressWarnings("unchecked")
+                        T response = (T) sb.toString();
```

Comment:
maybe using the StringUtils.toString(), which takes in an object will help in this case?